### DISCUSSION TITLE
Is there any to check whether the MIP solver (SCIP) is actually warm starting ?

### DISCUSSION THREAD

**Discussion by Regista6** [google/or-tools#3375] [Linear Solver questions] [answered]
Hello,

I am using the `SetHint` function ( `std::vector<std::pair<const MPVariable*, double> > hint`) to set the values of the decision variable as per the initial solution that I have.

I can't determine whether the solver is actually starting from this solution. Is there any way to check that?

Secondly, since I am warm starting, shouldn't the final objective value be at least equal to that of the initial solution?.  This doesn't seem to be the case.

Using or-tools v9.3 on VS 2022.

**Comment by lperron** [comment]
apart from enabling the solver log, I have no ideas.

**Reply by Regista6** [reply]
Hi,

Actually, after writing the question, I thought of enabling the solver log but couldn't find any syntax (C++) for it in the [documentation](https://developers.google.com/optimization/mip/mip_var_array#complete_programs)

How do I enable the solver log?

**Comment by lperron** [comment, marked as answer]
solver.EnableOutput()
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le ven. 8 juil. 2022 à 10:45, watchdogs132 ***@***.***> a
écrit :

**Comment by gregy4** [comment]
What about export of your model from ortools to format accepted by scip like lp ?  In case it works worse directly from scip, your hints work. In case their performance is same no hints are used.

**Reply by Regista6** [reply]
Hi,

I haven't tried that but upon turning on the solver log, I found that SCIP accepts my complete feasible solution (completesol heuristic) but is unable to improve the initial solution.

**Reply by gregy4** [reply]
It is maybe time to try another mip solver. Except commercial solvers I can recommend highs solver. It should be supported by ortools in future versions directly by highs team. So far you can check it via export to lp format.

**Reply by Regista6** [reply]
Thanks. Will try that.
